mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java

```diff
@@ -152,6 +153,13 @@
      */
     void deleteMailbox(MailboxPath mailboxPath, MailboxSession session) throws MailboxException;
 
+    /**
+     * Delete the mailbox with the given id
+     *
+     * @return the Mailbox when deleted
+     */
+    Mailbox deleteMailbox(MailboxId mailboxId, MailboxSession session) throws MailboxException;
```

Comment:
why do we want to return it?

Reply:
It turns out one of the caller needs the path to perform unsubscription.

Returning the mailbox solved a possible race condition here : 

```
A delete the mailbox in JMAP
B concurrently renames it
```

And 

```
A read (old) mailboxPath
B renames the mailbox to (new) mailboxPath
A deletes the mailbox by Id
A unsubsribe (old) mailboxPath instead of (new)

(new) remains an orphan subscribed method
```

Having deletes returning what had been deleted allows strongly consistant implementation that caller can use in a concurrently safe manner.

Reply:
> Having deletes returning what had been deleted allows strongly consistant implementation that caller can use in a concurrently safe manner.

Agree but this means that you cannot have an idempotent delete then. Just my 2 cents.

Reply:
> Agree but this means that you cannot have an idempotent delete then

AFAIK other factors prevent idempotence here:
 - At the API level, throwing when the mailbox do not exist
 - At the implementation publishing on the event bus (side effect)

Hence idempotence is not the goal I try to achieve here...